xls/integrator/ir_integrator.h

```diff
@@ -0,0 +1,55 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef XLS_INTEGRATOR_IR_INTEGRATOR_H_
+#define XLS_INTEGRATOR_IR_INTEGRATOR_H_
+
+#include "absl/status/statusor.h"
+#include "xls/ir/function.h"
+#include "xls/ir/package.h"
+
+namespace xls {
+
+class IntegrationBuilder {
```

Comment:
Class comment, plz!

Reply:
Also, based on the CL description itself, it's not immediately clear how an "integrated" function differs from a function after an inlining pass (that being said, after looking @ the CL, it's a lot clearer WRT maps/fors) - please be sure to add that info to this comment.

Reply:
Comment added.

The long term plan is to take 2 or more separte functions / circuits and create a single, reprogramamble circuit that can be configured to have the same functionality as the input circuits.

So if we input
func AXpB = A * X + b
and 
func AX = A*X

We can combine into something like
tmp = A*X
return config_signal ? tmp : tmp + b